### ISSUE TITLE
Ubuntu Mate not shown as install option when running wubi 16.04.2r319.exe in Win7

### ISSUE THREAD

**Issue by dieter-erich** [hakuna-m/wubiuefi#50]
Hi,
   I want to install ubuntu-mate 16.04.2 via wubi but this option does not show up in the GUI neither without nor with the corresponding iso file present in the same directory. I do not want to first install plain ubuntu 16.04.2 and then update/upgrade to mate. What might I do wrong?
Thanks, D-E

**Comment by hakuna-m** [comment]
If I run wubi16042r319.exe, it is possible to choose desktop environment  "Ubuntu MATE". But there is a pre-selection for local ISOs. The search path for local ISOs is:

- pre-defined ISOs with parameter isopath e.g. `wubi16042r319.exe --isopath=ubuntu-mate-16.04.2-desktop-amd64.iso`
-  same folder as wubi.exe
-  root directory of Windows drives e.g. `D:\`
-  Windows desktop

Note: If there is more than one ISO in the same folder, it pre-selects the first found file with suffix .iso which is not always the wanted ISO.

Maybe, that's the reason for your issue. But the log file in the user temp folder (`%temp%` on Windows) should log the real reason.

**Comment by dieter-erich** [comment]
Hi Hakuna,

   thanks a lot for your quick answer! I'll let you know whether it 
worked asap!

I hope that I do not bother you too much with one more question:

I had a working wubi installation of mate-16.04 (alongside with 
ubuntu-14.04 by following this method: 
https://askubuntu.com/questions/129149/how-to-enable-booting-multiple-ubuntu-wubi 
- but that is another story) on a win7 machine . Unfortunately, trying 
to clean up my wubi directories I lost access to my working installation 
(on booting into it I just get the splash screen but then it gets stuck).

I suppose that this is due to the presence of nvidia / cuda / drivers 
and/or the above intermingling of systems and/or the update to grub2.

My usual way to recover was - so far - to rename the ubuntu directory to 
e.g. "ubuntu_16.04_old", do a new installation of wubi (to get the 
correct boot files in place) and rename the old wubi directory back to 
'ubuntu'.  However, this time it did not work and I still get only the 
splash screen. Is there any (easy) way to recover my old installation 
that must be intact maybe except from the grub2 files?

When a new LTS version of Ubuntu becomes available I always keep the old 
one in parallel until the new one works as it should!

Thanks a lot, in particular for keeping wubi alive!

D-E

------------------------------------------------------------------------
Dieter Blaas,
Max F. Perutz Laboratories
Medical University of Vienna,
Inst. Med. Biochem., Vienna Biocenter (VBC),
Dr. Bohr Gasse 9/3,
A-1030 Vienna, Austria,
Tel: 0043 1 4277 61630,
Fax: 0043 1 4277 9616,
e-mail: dieter.blaas@meduniwien.ac.at
------------------------------------------------------------------------

Am 13.04.2017 um 19:05 schrieb hakuna matata:

**Comment by hakuna-m** [comment]
> Is there any (easy) way to recover my old installation that must be intact maybe except from the grub2 files?

If you move a working Wubi installation to a new folder, the file **/boot/grub/grub.cfg** within **root.disk** still contains the old folder name. So if you don't touch it within the new location, it is no problem to move it to the old folder.

If you boot into the Wubi installation at the new location and if you run e.g. `sudo update-grub`, the file **/boot/grub/grub.cfg** has been changed to the new folder name. i.e. If you move it to the old folder it is necessary to change the file **/boot/grub/grub.cfg** again.

An easy way to do so is to edit the GRUB menu. If the GRUB menu is not shown, press SHIFT key immediately after Windows menu. Then edit the first menu entry "Ubuntu" with `e`.

Change the new folder name e.g **ubuntu_1604** to **ubuntu** in the loopback line:
`loopback loop0 /ubuntu/disks/root.disk`
and in the linux line:
`linux	/vmlinuz root=UUID=YOUR-UUID-HERE loop=/ubuntu/disks/root.disk ro   quiet splash $vt_handoff`
Press `F10` to boot.

After successful boot, create a new **/boot/grub/grub.cfg** with
`sudo update-grub`

**Comment by aadrrshh** [comment]
I could install ubuntu Mate 17.04  in the following way. 

1. I unzipped the ubuntu-mate-17.04-desktop-amd64.iso file using 7z into a folder \folder\
2. I downloaded wubi16041r311.exe and put it into the folder from step 1
3. I double clicked the wubi exec and then I could see all the options. 

The only disadvantage was that it had to redownload the iso file. 

Cheers !

**Comment by hakuna-m** [comment]
@aadrrshh:
> 1. I unzipped the ubuntu-mate-17.04-desktop-amd64.iso file using 7z into a folder \folder\

IMHO, it makes no sense to unzip the iso. You can mount it as a virtual drive on Windows if you want no pre-selection but with the possibility to use it for Wubiuefi. **ubuntu-mate-17.04-desktop-amd64.iso** is desktop environment "Ubuntu MATE" for a wubi**1704**r....exe or "Wubi EFI" for a not matching wubi.exe 

> I downloaded wubi16041r311.exe and put it into the folder from step 1

wubi**16041**r....exe is for **16.04.1**. It is possible to use a 64 bit 17.04 ISO as "Wubi EFI", but if you select "Ubuntu MATE" it downloads ubuntu-mate-**16.04.1**-desktop-amd64.iso
